### ISSUE TITLE
amp need a whitelist

### ISSUE THREAD

**Issue by chenkoucer** [pytorch/pytorch#117832]
### 🚀 The feature, motivation and pitch

Can we provide a whitelist mechanism for amp features so that users can freely control the precision of the operator。
For example, I suspect that a operator running with low_precision_fp might affect the  accuracy, so I want to put it on fp32, but I don't find a good way to verify it.
...
with torch.cuda.amp.autocast(dtype=fp16, white_list={op_xx, op_yy}):
            outputs = model(inputs)
            loss = criterion(outputs, targets)

I want all the operators in  white_list={op_xx, op_yy} will run in fp32.  
Are you willing to provide such a variable  in amp interface for users,  thanks much!
https://pytorch.org/docs/stable/amp.html#module-torch.cuda.amp
